configtype/duration.go

```diff
@@ -19,6 +21,15 @@ func NewDuration(d time.Duration) Duration {
 	}
 }
 
+func (Duration) JSONSchema() *jsonschema.Schema {
+	// provide a pattern for only non-negative durations
```

Comment:
The regex doesn't look like it only supports non-negative durations?

Reply:
yeah, I'll remove the comment (as we do support negative durations even in tests)

Reply:
removed in df0b27e